### PR TITLE
Articles are displayed by category

### PR DESCRIPTION
**Pull request by alghazu**
### What we did: 
- Add category model and controller
- Create relation between category and article models
- Create a nav bar for articles
- Create a drop down menu for categories when creating new article
- Modify previous tests including article creation to include category attribute

### CODE REVIEW

**Comment by alghazu** [comment]
We did, there is a feature file created in first commit `c90bcd2`
I guess you mean to create a separate commit after creating the feature file?
Next PR :D  @tochman

**Review by ghost — APPROVED** [review]
Looks good to me

**Comment by alghazu** [comment]
@tochman There are specs for category.
https://github.com/CraftAcademy/newsroom-april/pull/10/files#diff-2dd1d3edb24becaa29be930973f468e5

File: app/views/articles/index.html.haml [hunk 1]
```diff
@@ -1,4 +1,10 @@
 %h1 News Room
+#category-bar
+  = link_to "All", root_path
+  - Category.all.each do |category|
```
Comment (tochman) [inline app/views/articles/index.html.haml]: Please avoid calling models from a view. Set that in the controller. `@categories = Category.all`

File: app/views/articles/new.html.haml [hunk 2]
```diff
@@ -9,4 +9,8 @@
   %br
   = form.text_field :content
   %br
+  = form.label :category
+  %br
+  = form.collection_select :category_id, Category.all, :id, :name
```
Comment (tochman) [inline app/views/articles/new.html.haml]: Please avoid calling models from a view. Set that in the controller or a helper. `@categories = Category.all`

File: config/routes.rb [hunk 3]
```diff
@@ -1,4 +1,5 @@
 Rails.application.routes.draw do
   resources :articles, only: [:new, :create, :show]
+  resources :categories, only: [:show]
```
Comment (tochman) [inline config/routes.rb]: When do you use the `show` route?

File: spec/models/article_spec.rb [hunk 4]
```diff
@@ -11,7 +11,7 @@
   describe 'Validation' do
     it { is_expected.to validate_presence_of :headline }
     it { is_expected.to validate_presence_of :content }
-
+    it { is_expected.to belong_to :category}
```
Comment (tochman) [inline spec/models/article_spec.rb]: In what way is this a validation test?

File: spec/models/category_spec.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  end
+
+  describe 'Validation' do
+    it { is_expected.to validate_presence_of :name }
+    it { is_expected.to have_many :articles }
+  end
+
+  describe 'Factory' do
+    it 'can create a valid instance' do
+      expect(category).to be_valid
```
Comment (tochman) [inline spec/models/category_spec.rb]: Please take a look at the other model specs and factory tests. Check for consistency. Are you following the same format throughout the project?

**Review by tochman — CHANGES_REQUESTED** [review]
Please make sure that your feature file names reflect what the feature is about. `create_article_form.feature` and `article_categories.feature` feels weird.

File: app/views/articles/index.html.haml (hunk 1 above)
Comment (alghazu) [reply app/views/articles/index.html.haml]: :+1:

File: app/views/articles/new.html.haml (hunk 2 above)
Comment (alghazu) [reply app/views/articles/new.html.haml]: :+1:

File: config/routes.rb (hunk 3 above)
Comment (alghazu) [reply config/routes.rb]: When a category is clicked from categories-bar, it links to the category-show route, which displays all the articles in that category on index page.
Should we use a different route, e.g. call it (filter) or so?

File: spec/models/article_spec.rb (hunk 4 above)
Comment (alghazu) [reply spec/models/article_spec.rb]: :+1:

File: spec/models/category_spec.rb (hunk 5 above)
Comment (alghazu) [reply spec/models/category_spec.rb]: :+1:

**Comment by alghazu** [comment]
@tochman  All comments addressed, and solved merge conflicts

**Pull request merged**